projects/js-packages/components/components/scan-report/index.tsx

```diff
@@ -0,0 +1,208 @@
+							return (
+								<Tooltip
+									className={ styles.tooltip }
+									text={ __( 'Threat detected.', 'jetpack-components' ) }
+								>
+									<div className={ styles.icon }>
+										<ShieldIcon variant="warning" height={ iconHeight } />
+									</div>
+								</Tooltip>
+							);
+						}
+						return (
+							<Tooltip
+								className={ styles.tooltip }
+								text={ __(
+									'No known threats found that affect this version.',
+									'jetpack-components'
+								) }
+							>
+								<div className={ styles.icon }>
+									<ShieldIcon variant="success" height={ iconHeight } />
+								</div>
+							</Tooltip>
+						);
+					}
+					return (
+						<Tooltip
+							className={ styles.tooltip }
+							text={ __(
+								'This item was added to your site after the most recent scan. We will check for threats during the next scheduled one.',
+								'jetpack-components'
+							) }
+						>
+							<div className={ styles.icon }>
+								<ShieldIcon variant="info" height={ iconHeight } />
+							</div>
+						</Tooltip>
+					);
+				},
+			},
+			{
+				id: FIELD_TYPE,
+				label: __( 'Type', 'jetpack-components' ),
+				elements: TYPES,
+				render( { item } ) {
+					return item.type ? item.type.charAt( 0 ).toUpperCase() + item.type.slice( 1 ) : '';
+				},
+			},
+			{
+				id: FIELD_NAME,
+				label: __( 'Name', 'jetpack-components' ),
+				enableGlobalSearch: true,
+				render( { item } ) {
+					return item.name ? item.name : '';
+				},
+			},
+			{
+				id: FIELD_VERSION,
+				label: __( 'Version', 'jetpack-components' ),
+				enableGlobalSearch: true,
+				render( { item } ) {
+					return item.version ? item.version : '';
+				},
+			},
+			{
+				id: FIELD_ICON,
+				label: __( 'Icon', 'jetpack-components' ),
+				render( { item } ) {
+					return (
+						<div className={ styles.threat__media }>
+							<Icon icon={ ICONS[ item.type ] } />
+						</div>
+					);
+				},
+			},
+		];
+
+		return result;
+	}, [] );
+
+	/**
+	 * Apply the view settings (i.e. filters, sorting, pagination) to the dataset.
+	 *
+	 * @see https://github.com/WordPress/gutenberg/blob/trunk/packages/dataviews/src/filter-and-sort-data-view.ts
+	 */
+	const { data: processedData, paginationInfo } = useMemo( () => {
+		return filterSortAndPaginate( data, view, fields );
+	}, [ data, view, fields ] );
+
+	/**
+	 * Callback function to update the view state.
+	 *
+	 * @see https://developer.wordpress.org/block-editor/reference-guides/packages/packages-dataviews/#onchangeview-function
+	 */
+	const onChangeView = useCallback( ( newView: View ) => {
+		setView( newView );
+	}, [] );
+
+	/**
+	 * DataView getItemId function - returns the unique ID for each record in the dataset.
+	 *
+	 * @see https://developer.wordpress.org/block-editor/reference-guides/packages/packages-dataviews/#getitemid-function
+	 */
+	const getItemId = useCallback( ( item: Threat ) => item.id.toString(), [] );
```

Comment:
```suggestion
	const getItemId = useCallback( ( item: Extension ) => item.slug, [] );
```

Reply:
How I've got this to work in Protect is by passing spreading the `core`, `plugins`, `themes`, and a modified `files` variable from the `useProtectData` hook to this component. The `core` and `files` variables will no include a `slug` so instead I've just combined the set and indexed them like so:
```
	const {
		results: { core, plugins, themes, files },
	} = useProtectData();

	const data = [
		...core,
		...plugins,
		...themes,
		{ checked: true, threats: files, type: 'files' },
	].map( ( item, index ) => {
		return { id: index + 1, ...item };
	} );
```

Obviously the `Threat` doesn't apply here, but `Extension` won't either. Any thought on how we could approach this differently?

Reply:
(We could add a custom type if `Extension` is not applicable, i.e. `ScanReportExtension`)

Reply:
If you do go with generating an ID, I would recommend using the slug or any other unique identifiers whenever possible as opposed to index.

Reply:
>The useProtectData hook is being removed in add/protect/core, we will be referencing data from protect-status here which should return the full Extension.

Ah good to know, I missed that entirely. How will file `core` and `file` threats be formatted? Will all follow the same structure?

Reply:
They should remain the same, with `core` being an `Extension` with threats, and `files` being an array of file threats.

Reply:
Added/applied a custom `ScanReportExtension` type. I've kept with generating an ID for now, because for `core` the `slug` will always be `null`, and for the variable we'll have to generate from `files` and pass to the component, it seems a little awkward to predefine in someway. Perhaps we can add a followup to investigate an improved way to use unique identifier, unless you have a different recommendation?